tracer/src/Datadog.Trace/DataStreamsMonitoring/Aggregation/DataStreamsAggregator.cs

```diff
@@ -149,7 +149,8 @@ internal void Clear(List<SerializableStatsBucket> statsToRemove)
                 hash: point.Hash,
                 parentHash: point.ParentHash,
                 pathwayLatency: _sketchPool.Get(),
-                edgeLatency: _sketchPool.Get());
+                edgeLatency: _sketchPool.Get(),
+                payloadSize: _sketchPool.Get());
```

Comment:
You need to make sure to call `_sketchPool.Release(bucket.Value.PayloadSize);` in `Clear()`